erpnext/stock/doctype/stock_ledger_entry/stock_ledger_entry.py

```diff
@@ -96,6 +97,12 @@ def check_stock_frozen_date(self):
 	def scrub_posting_time(self):
 		if not self.posting_time or self.posting_time == '00:0':
 			self.posting_time = '00:00'
+			
+	def validate_batch(self):
+		if self.batch_no:
```

Comment:
and self.voucher_type != "Stock Entry"

For stock entry it is validated selectively in its controller based on purpose